community/kernel/src/test/java/org/neo4j/kernel/impl/proc/ReflectiveProcedureTest.java

```diff
@@ -305,6 +309,65 @@ public void shouldSupportProcedureDeprecation() throws Throwable
         }
     }
 
+    @Test
+    public void shouldLoadWhiteListedProcedure() throws Throwable
+    {
+        // Given
+        ProcedureConfig config = new ProcedureConfig( Config.defaults().with(
+                genericMap( procedure_white_list.name(), "org.neo4j.kernel.impl.proc.listCoolPeople") ) );
+
+        Log log = mock(Log.class);
+        ReflectiveProcedureCompiler procedureCompiler = new ReflectiveProcedureCompiler( new TypeMappers(), components,
+                components, log, config );
+
+        // When
+        CallableProcedure proc =
+                procedureCompiler.compileProcedure( SingleReadOnlyProcedure.class, Optional.empty(), false ).get( 0 );
+        // When
+        RawIterator<Object[],ProcedureException> result = proc.apply( new BasicContext(), new Object[0] );
+
+        // Then
+        assertEquals( result.next()[0], "Bonnie" );
+    }
+
+    @Test
+    public void shouldNotLoadNoneWhiteListedProcedure() throws Throwable
+    {
+        // Given
+        ProcedureConfig config = new ProcedureConfig( Config.defaults().with(
+                genericMap( procedure_white_list.name(), "org.neo4j.kernel.impl.proc.NOTlistCoolPeople") ) );
+
+        Log log = mock(Log.class);
+        ReflectiveProcedureCompiler procedureCompiler = new ReflectiveProcedureCompiler( new TypeMappers(), components,
+                components, log, config );
+
+        // When
+        List<CallableProcedure> proc =
+                procedureCompiler.compileProcedure( SingleReadOnlyProcedure.class, Optional.empty(), false );
+        // Then
+        verify( log )
+                .warn( "The procedure 'org.neo4j.kernel.impl.proc.listCoolPeople' is not white listed." );
+        assertThat( proc.isEmpty(), is(true) );
+    }
+
+    @Test
+    public void shouldLoadFullAccessProcedure() throws Throwable
```

Comment:
This doesn't seem to test what it says, and looks very similar to `shouldLoadWhiteListedProcedure `